### PR TITLE
Reverted internal prefix for every backend except Dafny

### PR DESCRIPTION
**Pull request by MikaelMayer**
### Description
Some clients using Smithy-Dafny were stuck because we changed the internal field prefix. We revert it in this PR by convenience for them as we don't have an immediate need for this enhanced prefix.

### How has this been tested?
All tests should pass.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public abstract partial class SinglePassCodeGenerator {
     // we can prefix it with "_".
     // However, for backends such as Rust which need special internal fields, we want to clearly
     // disambiguate between compiler-generated names and user-defined names, hence this prefix.
```
Comment (robin-aws) [inline Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs:31]: Not worth extra CI runs to fix, but sounds like this comment isn't accurate any more?

**Review by robin-aws — APPROVED** [review]
LGTM, although where is the second definition of this prefix that the Dafny-based backends are still using?

File: Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs (hunk 1 above)
Comment (MikaelMayer) [reply Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs:31]: Not completely accurate, but still indicates why this prefix isn't hard-coded at least.

File: Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs [hunk 2]
```diff
@@ -27,9 +27,8 @@ static class SinglePassCompilerExtensions {
   public abstract partial class SinglePassCodeGenerator {
     // Dafny names cannot start with "_". Hence, if an internal Dafny name is problematic in the target language,
     // we can prefix it with "_".
-    // However, for backends such as Rust which need special internal fields, we want to clearly
-    // disambiguate between compiler-generated names and user-defined names, hence this prefix.
-    protected virtual string InternalFieldPrefix => "_i_";
+    // This prefix can be overridden as necessary by backends.
```
Comment (robin-aws) [inline Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.cs:30]: Thanks :)

**Review by robin-aws — APPROVED** [review]

**Pull request merged**